### PR TITLE
Demo_Effect

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_Demo_Effect/z_demo_effect.h [hunk 1]
```diff
@@ -6,10 +6,20 @@
...
+typedef void (*DemoEffectUnkFunc2)(Actor*, GlobalContext*);
+
+#define DEMOEFFECT_GET_FF(thisx) ((thisx)->params & 0xFF)
 
 typedef struct DemoEffect {
     /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x34];
+    /* 0x0144 */ SkelAnimeCurve skelCurve;
+    /* 0x0164 */ u8 unk_164;
+    /* 0x0165 */ UNK_TYPE1 unk165[3];
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Demo_Effect/z_demo_effect.h:18]:
```suggestion
    /* 0x0165 */ UNK_TYPE1 unk_165[3];
```

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_Demo_Effect/z_demo_effect.c [hunk 2]
```diff
@@ -32,33 +37,284 @@ const ActorInit Demo_Effect_InitVars = {
...
+    return true;
+}
+
+void func_808CDFF8(Actor* thisx, GlobalContext* globalCtx) {
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    POLY_XLU_DISP = Gfx_CallSetupDL(POLY_XLU_DISP, 25);
+
+    Matrix_Scale(2.0f, 2.0f, 2.0f, MTXMODE_APPLY);
+    SkelCurve_Draw(thisx, globalCtx, &THIS->skelCurve, func_808CDE78, NULL, 1, thisx);
```
Comment (hensldm) [inline src/overlays/actors/ovl_Demo_Effect/z_demo_effect.c:280]:
```suggestion
    SkelCurve_Draw(thisx, globalCtx, &THIS->skelCurve, func_808CDE78, NULL, 1, thisx);
```
Not enough room on the stack for this cast, or is something push too far down?

Comment (Kelebek1) [reply src/overlays/actors/ovl_Demo_Effect/z_demo_effect.c:280]: Pushed down by 4: https://decomp.me/scratch/ot2Fv

Comment (hensldm) [reply src/overlays/actors/ovl_Demo_Effect/z_demo_effect.c:280]: https://decomp.me/scratch/TUU8D, seems to work with a `gfxCtx` temp which I think is preferable.

**Review by hensldm — APPROVED** [review]
LGTM, though Jenkins seems to be mad about a bad checksum, bss?

**Review by Kenix3 — APPROVED** [review]
The system at a high level looks similar to OOTs actor with the same name. Seems like a lot of sub systems were removed.

**Pull request merged**